projects/utils-browser/src/lib/layout/colors.ts

```diff
@@ -0,0 +1,177 @@
+
```

Comment:
For this we could use a small library e.g. https://gka.github.io/chroma.js so we don't have to reimplement all the stuff.